sys-cluster/glusterfs/glusterfs-9.2.ebuild

```diff
@@ -0,0 +1,195 @@
+		$(use_enable test cmocka) \
+		$(use_enable xml xml-output) \
+		$(use libtirpc || echo --without-libtirpc) \
+		$(use ipv6 && echo --with-ipv6-default) \
+		--with-tmpfilesdir="${EPREFIX}"/usr/lib/tmpfiles.d \
+		--localstatedir="${EPREFIX}"/var
+}
+
+src_compile() {
+	default
+	use emacs && elisp-compile extras/glusterfs-mode.el
+}
+
+src_test() {
+	./run-tests.sh || die
+}
+
+src_install() {
+	default
+
+	rm \
+		"${ED}"/etc/glusterfs/glusterfs-{georep-,}logrotate \
+		"${ED}"/etc/glusterfs/gluster-rsyslog-*.conf \
+		"${ED}"/usr/share/doc/${PF}/glusterfs{-mode.el,.vim} || die "removing false files failed"
+
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/glusterfs.logrotate glusterfs
+
+	if use rsyslog ; then
+		insinto /etc/rsyslog.d
+		newins extras/gluster-rsyslog-7.2.conf 60-gluster.conf
+	fi
+
+	if use emacs ; then
+		elisp-install ${PN} extras/glusterfs-mode.el*
+		elisp-site-file-install "${FILESDIR}/${SITEFILE}"
+	fi
+
+	insinto /usr/share/vim/vimfiles/ftdetect; doins "${FILESDIR}"/${PN}.vim
+	insinto /usr/share/vim/vimfiles/syntax; doins extras/${PN}.vim
+
+	# insert some other tools which might be useful
+	insinto /usr/share/glusterfs/scripts
+	doins \
+		extras/backend-{cleanup,xattr-sanitize}.sh \
+		extras/clear_xattrs.sh \
+		extras/migrate-unify-to-distribute.sh
+
+	# correct permissions on installed scripts
+	# fperms 0755 /usr/share/glusterfs/scripts/*.sh
+	chmod 0755 "${ED}"/usr/share/glusterfs/scripts/*.sh || die
+
+	newinitd "${FILESDIR}/${PN}-r1.initd" glusterfsd
+	newinitd "${FILESDIR}/glusterd-r3.initd" glusterd
+	newconfd "${FILESDIR}/${PN}.confd" glusterfsd
+
+	keepdir /var/log/${PN}
+	keepdir /var/lib/glusterd/{events,glusterfind/.keys}
+
+	# QA
+	rm -r "${ED}/var/run/" || die
+	if ! use static-libs; then
+		find "${D}" -type f -name '*.la' -delete || die
+	fi
+
+	python_optimize "${ED}"
+}
+
+pkg_postinst() {
+	elog "Starting with ${PN}-3.1.0, you can use the glusterd daemon to configure your"
+	elog "volumes dynamically. To do so, simply use the gluster CLI after running:"
+	elog "  /etc/init.d/glusterd start"
+	echo
+	elog "For static configurations, the glusterfsd startup script can be multiplexed."
+	elog "The default startup script uses /etc/conf.d/glusterfsd to configure the"
+	elog "separate service.  To create additional instances of the glusterfsd service"
+	elog "simply create a symlink to the glusterfsd startup script."
+	echo
+	elog "Example:"
+	elog "    # ln -s glusterfsd /etc/init.d/glusterfsd2"
+	elog "    # ${EDITOR} /etc/glusterfs/glusterfsd2.vol"
+	elog "You can now treat glusterfsd2 like any other service"
+	echo
+	ewarn "You need to use a ntp client to keep the clocks synchronized across all"
+	ewarn "of your servers. Setup a NTP synchronizing service before attempting to"
+	ewarn "run GlusterFS."
+	echo
+	elog "If you are upgrading from a previous version of ${PN}, please read:"
+	elog "  http://docs.gluster.org/en/latest/Upgrade-Guide/upgrade_to_$(ver_cut '1-2')/"
+
+	tmpfiles_process
```

Comment:
I'm getting 
```
 * ERROR: sys-cluster/glusterfs-9.2::gentoo failed (postinst phase):
 *   tmpfiles_process: Must specify at least one filename
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called pkg_postinst
 *   environment, line 2230:  Called tmpfiles_process
 *   environment, line 3437:  Called die
 * The specific snippet of code:
 *       [[ ${#} -gt 0 ]] || die "${FUNCNAME}: Must specify at least one filename";
```
In every run, with 8.5 and 9.2. I can't see how this would be your fault, but... it doesn't look like this call is doing much in here. at least for me. @ionenwks care to test too?

Reply:
Before it was already installing tmpfiles.d files but without processing them. This `tmpfiles_process` is new and doesn't have automagic to find new .conf files, so they need to be specified.

If a single gluster.conf the only thing it can install, then this should be simple.

On a side-note, the init script seems to still be using /var/run/gluster while .conf is /run/gluster (it's normally a symlink but who knows with users' systems).

Reply:
I'm a little uncertain of this particular line:

```
101         --with-tmpfilesdir="${EPREFIX}"/usr/lib/tmpfiles.d \
```

(The normal question of when to specify EPREFIX in paths and when not to ... --localstatedir (next line) was with, so I followed that.  If someone can please confirm.

I've fixed the other stuff I believe.

Reply:
> ```
> 101         --with-tmpfilesdir="${EPREFIX}"/usr/lib/tmpfiles.d \
> ```
Seeing how it's a top level path (/usr), it'd mean it's not adding what was passed to --prefix= and so you need it.

And sure enough there's no issue and I'm getting:
`home/ionen/prefix/usr/lib/tmpfiles.d/gluster.conf`

However, ***if*** you want to worry about prefix, there's also this:
```
 * QA Notice: the following files are outside of the prefix:
 * /sbin
 * /sbin/mount.glusterfs
 * ERROR: sys-cluster/glusterfs-9.2::gentoo failed:
 *   Aborting due to QA concerns: there are files installed outside the prefix
```

Reply:
That /sbin is an interesting one.  This comes from the package itself I guess, if you don't mind let's let that one slip for the moment.  Not really looking forward to digging into that particular code, but we can report an issue upstream perhaps?